verticapy/learn/memmodel.py

```diff
@@ -0,0 +1,1036 @@
+#            |______      /    /     /
+#                   |____/    /     /
+#          _____________     /     /
+#          \           /    /     /
+#           \         /    /     /
+#            \_______/    /     /
+#             ______     /     /
+#             \    /    /     /
+#              \  /    /     /
+#               \/    /     /
+#                    /     /
+#                   /     /
+#                   \    /
+#                    \  /
+#                     \/
+#                    _
+# \  / _  __|_. _ _ |_)
+#  \/ (/_|  | |(_(_|| \/
+#                     /
+# VerticaPy is a Python library with scikit-like functionality to use to conduct
+# data science projects on data stored in Vertica, taking advantage Vertica’s
+# speed and built-in analytics and machine learning features. It supports the
+# entire data science life cycle, uses a ‘pipeline’ mechanism to sequentialize
+# data transformation operations, and offers beautiful graphical options.
+#
+# VerticaPy aims to solve all of these problems. The idea is simple: instead
+# of moving data around for processing, VerticaPy brings the logic to the data.
+#
+#
+# Modules
+#
+# Standard Python Modules
+import numpy as np
+
+# VerticaPy Modules
+from verticapy.toolbox import *
+from verticapy.errors import *
+
+# ---#
+def predict_from_coef(X: Union[list, np.array], 
+                      coefficients: Union[list, np.array], 
+                      intercept: float, 
+                      method: str = "LinearRegression",
+                      return_proba: bool = False,):
+    """
+    ---------------------------------------------------------------------------
+    Predicts using a linear regression model and the input attributes.
+
+    Parameters
+    ----------
+    X: list / numpy.array
+        Data on which to make the prediction.
+    coefficients: list / numpy.array
+        List of the model's coefficients.
+    intercept: float
+        The intercept or constant value.
+    method: str, optional
+        The model category, one of the following: 'LinearRegression', 'LinearSVR', 
+        'LogisticRegression', or 'LinearSVC'.
+    return_proba: bool, optional
+        If set to True and the method is set to 'LogisticRegression' or 'LinearSVC', 
+        the probability is returned.
+
+    Returns
+    -------
+    numpy.array
+        Predicted values
+    """
+    check_types([("X", X, [list,],), 
+                 ("coefficients", coefficients, [list,],),
+                 ("intercept", intercept, [float, int,],),
+                 ("method", method, ["LinearRegression", "LinearSVR", "LogisticRegression", "LinearSVC"],),
+                 ("return_proba", return_proba, [bool],),])
+    result = intercept + np.sum(np.array(coefficients) * np.array(X), axis=1)
+    if method in ("LogisticRegression",):
+        result = 1 / (1 + np.exp(- (result)))
+    elif method in ("LinearSVC",):
+        result =  1 - 1 / (1 + np.exp(result))
```

Comment:
You may notice that the formulas are the equal; i.e., 1 / (1 + np.exp(- (result))) = 1 - 1 / (1 + np.exp(result)). So, you can consolidate the if statements.

Reply:
Done;